app/src/lib/git/status.ts

```diff
@@ -187,3 +165,96 @@ export async function getStatus(
     workingDirectory,
   }
 }
+
+/**
+ *
+ * Update map of working directory changes with a file status entry.
+ * Reducer(ish).
+ *
+ * (Map is used here to maintain insertion order.)
+ */
+function buildStatusMap(
+  files: Map<string, WorkingDirectoryFileChange>,
+  entry: IStatusEntry,
+  filesWithConflictMarkers: Set<string>
+): Map<string, WorkingDirectoryFileChange> {
+  const status = mapStatus(entry.statusCode)
+
+  if (status.kind === 'ordinary') {
+    // when a file is added in the index but then removed in the working
+    // directory, the file won't be part of the commit, so we can skip
+    // displaying this entry in the changes list
+    if (
```

Comment:
This can be combined with the `if` statement above to reduce the nesting.

Reply:
mind if I make a followup issue to refactor getStatus more? the above line was only a whitespace change, but i'd love to dig in some more :)

Reply:
#5796